migrations/emailmonitor/oxa_email_monitor.py

```diff
@@ -0,0 +1,215 @@
+# This python script determines the failed activation email sends and also creates daily statistics in database
+# This summary data and email failures used in another script for sending status reports and also for sending activation emails again
+
+import MySQLdb
+import collections
+import os
+import re
+from datetime import datetime,timedelta
+
+# Write an error or info log line with timestamp to the application log file /oxa/oxa_mail_monitor.log
+def write_log( log, error=None ):
+    if error == None:
+        os.system("echo '"+str(datetime.now())+" "+ log + "' >> /oxa/oxa_email_monitor.log")
+    else:
+        os.system("echo '"+str(datetime.now())+" "+ log + ": "+str(error)+"' >> /oxa/oxa_email_monitor.log")
+		
+# This function creates the YYYY,MM,DD statistics raw in oxa.oxa_activationsummary table if it doesn't exists. This is rerunnable
+def create_year_month_day():
+    db = MySQLdb.connect(IP,User,Password,Database)
+    cursor = db.cursor()
+
+    sql = "SELECT YEAR(date_joined), MONTH(date_joined), DAY(date_joined), COUNT(*) FROM edxapp.auth_user \
+          GROUP BY DATE(date_joined) \
+          ORDER BY DATE(date_joined)"
+
+    cursor.execute(sql)
+    results = cursor.fetchall()
+    
+    for row in results:
+        y = str(row[0])
+        m = str(row[1])
+        d = str(row[2])
+        sql2 = "SELECT id FROM oxa.oxa_activationsummary WHERE xyear="+y+" and xmonth="+m+" and xday="+d
+        cursor.execute(sql2)
+        if cursor.rowcount < 1: 
```

Comment:
Common Elton. For even 10 years we are talking about 3650 rows. This is nothin for MySQL. Also I created index on YYYY+MM+DD. No performance problem here.